### PR TITLE
fix: use our version of `fast-stable-stringify` everywhere

### PR DESCRIPTION
**Pull request by steveluscher**

### CODE REVIEW

**Comment by steveluscher** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2504?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2504** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2504?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2503** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2503?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2502** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2502?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2501** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2501?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @steveluscher and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/library-legacy/src/connection.ts [hunk 1]
```diff
@@ -6291,10 +6290,7 @@
     args: IWSRequestParams,
   ): ClientSubscriptionId {
     const clientSubscriptionId = this._nextClientSubscriptionId++;
-    const hash = fastStableStringify(
-      [subscriptionConfig.method, args],
-      true /* isArrayProp */,
```
Comment (steveluscher) [inline packages/library-legacy/src/connection.ts:6296]: Pretty sure this never did anything. TypeScript passed because this function used to be `any`.

**Review by lorisleiva — APPROVED** [review]
Love the `@solana/*` only dependencies!

**Comment by steveluscher** [comment]
### Merge activity

* **Apr 17, 12:36 AM PDT**: @steveluscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2504).

**Comment by ZhengYuTay** [comment]
@steveluscher seems like it cause npm installed to fail, the dep wasnt replaced with a published version when published
image

**Comment by solanamonk** [comment]
I believe this PR broke the package.

```
$ pnpm update @solana/web3.js
 ERR_PNPM_WORKSPACE_PKG_NOT_FOUND  In : "@solana/fast-stable-stringify@workspace:*" is in the dependencies but no package named "@solana/fast-stable-stringify" is present in the workspace

This error happened while installing the dependencies of @solana/web3.js@1.91.5

Packages found in the workspace: 
Progress: resolved 68, reused 69, downloaded 0, added 0
```

**Comment by steveluscher** [comment]
Thanks for the report. 1.91.5 is deprecated now. Apparently `semantic-release` doesn't use or resolve `pnpm` workspace references when publishing.

**Pull request merged**